LabProject/src/main/java/agh/ics/oop/Animal.java

```diff
@@ -1,48 +1,60 @@
 package agh.ics.oop;
 
 public class Animal {
+    private static final Vector2d LOWER_BOUND = new Vector2d(0,0);
+    private static final Vector2d UPPER_BOUND = new Vector2d(4,4);
     private MapDirection direction;
-    private Vector2d location;
+    private IWorldMap map;
+    private Vector2d position;
+    public Animal(IWorldMap map)
```

Comment:
można to jeszcze zwinąć tak by mniejszy konstruktor wołał większy. Wtedy zauważymy np. że coś trzeba zrobić z `position`, bo inaczej będzie ustawione na null (a możemy ustawić je na to defaultowe jak wcześniej)